apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsTaggedControl.tsx

```diff
@@ -0,0 +1,71 @@
+import { FormControl, VStack } from "@chakra-ui/react"
+import {
+  and,
+  ControlProps,
+  RankedTester,
+  rankWith,
+  schemaMatches,
+} from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+import { FormLabel, MultiSelect } from "@opengovsg/design-system-react"
+import { ArticlePagePageProps } from "@opengovsg/isomer-components"
+
+import { JSON_FORMS_RANKING } from "~/constants/formBuilder"
+import { collectionItemSchema } from "~/features/editing-experience/schema"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { trpc } from "~/utils/trpc"
+
+export const jsonFormsTaggedControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.ArrayControl,
+  and(schemaMatches((schema) => schema.format === "tagged")),
+)
+
+interface TaggedControlProps extends Omit<ControlProps, "data"> {
+  data: ArticlePagePageProps["tagged"]
```

Comment:
thought: do you think it would be worth to not do this, but rather, stick to using `ControlProps` and in the `JsonFormsTaggedControl` renderer we use a type predicate function to do the narrowing to `ArticlePagePageProps["tagged"]`?

Reason is because while we control the schema and the tester for this form control, in theory it is possible for all types of pages to be using this form control (since they all come from the generic `<FormBuilder>` component. We are doing this for specific cases that have this field, so wanted to be extra safe in the handling of the data.

(don't **have** to do it btw, just raising a thought)

Reply:
As discussed offline, not worth doing now